### PR TITLE
internationalized login errors

### LINKED ISSUE #8967: Login errors not localized
### Problem

Login errors are only displayed in English, regardless of chosen language.

#### Evidence / Screenshot
localized_login_errors
_Highlighted area displays non-localized strings._

#### Relevant URL(s)
https://openlibrary.org/account/login

### Reproducing the bug

1. Go to the `/account/login` page
2. Select a language other than English from the language dropper, found on the page's banner
3. Log in with invalid credentials

* Expected behavior: Error message is written in the selected language
* Actual behavior: Error message is English

### Context

- Browser (Chrome, Safari, Firefox, etc): All
- OS (Windows, Mac, etc): All
- Logged in (Y/N): N
- Environment (prod, dev, local): prod

#### Proposal & constraints

Replace hard-coded login errors with localized strings.

#### Related files

The error strings can be found in the `LOGIN_ERRORS` `dict`, here:
https://github.com/internetarchive/openlibrary/blob/a2f111c8cf6b52e2e30f7ae1fb009b76d3c92b5e/openlibrary/plugins/upstream/account.py#L60-L76

Keys of this `dict` should remain as they are --- only the values should change.

#### Stakeholders

@rebecca-shoptaw

### PR DESCRIPTION
**Pull request by etgibbs**
Closes #8967
@RBrar66 and I worked on this fix together, he was struggling w/ build set up. we go to the same school.

should fix the described issue with non-localized error messages, following the example of @rebecca-shoptaw 

### Technical

we thought that simply changing the formatting of the strings would be enough, but ran into this error:
image
we were able to pass test cases by adding an empty fallback value `load_translations(getattr(web.ctx, 'lang', ''))`
please let me know if this is the correct approach, or why we would be getting this error when attempting to format the strings in the `LOGIN_ERRORS` dictionary.

in addition, the strings in the dictionary auto-formatted on commit - I am assuming this is how they are supposed to be according to the style guide but I thought it was more readable being on one line. 

### Testing

running `docker compose run --rm home make test` without the proposed change  to `__init__.py` and the modified dictionary should result in the same before/after results.

### Stakeholders

@rebecca-shoptaw

### CODE REVIEW

**Comment by rebecca-shoptaw** [comment]
@etgibbs & @RBrar66 - Update! I came up with a nice workaround that prevents the errors but doesn't require altering `__init__.py`.

Looks like the issue was just that the translation was called at the top of the file, before `web.ctx` was fully loaded, so the fix just involves moving the dictionary into a function, as in `mybooks.py`.

In this case, because we want the return value to be accessible to many sub-functions but don't want to call the function at the top of the file, the implementation can be as follows:

1. Nest the login errors in a function that returns the specific error, i.e.:
```
def get_login_error(error_key):
    LOGIN_ERRORS = {
        "invalid_email": _("The email address you entered is invalid"),
        "account_blocked": _("This account has been blocked"),
        "account_locked": _("This account has been blocked"),
        ...
    }

    return LOGIN_ERRORS[error_key]
```

2. Search the file for references to `LOGIN_ERRORS` and replace each with calls to the new function, i.e.:
```
except ValueError:
                f.note = get_login_error('max_retries_exceeded')
```

P.S. Looks like there's a typo in one of the errors -- feel free to change the "account_locked" message to say "locked" instead of "blocked"

**Review by rebecca-shoptaw — CHANGES_REQUESTED** [review]
@etgibbs & @RBrar66 - Update! I came up with a nice workaround that prevents the errors but doesn't require altering `__init__.py`.

Looks like the issue was just that the translation was called at the top of the file, before `web.ctx` was fully loaded, so the fix just involves moving the dictionary into a function, as in `mybooks.py`.

In this case, because we want the return value to be accessible to many sub-functions but don't want to call the function at the top of the file, the implementation can be as follows:

1. Nest the login errors in a function that returns the specific error, i.e.:
```
def get_login_error(error_key):
    LOGIN_ERRORS = {
        "invalid_email": _("The email address you entered is invalid"),
        "account_blocked": _("This account has been blocked"),
        "account_locked": _("This account has been blocked"),
        ...
    }

    return LOGIN_ERRORS[error_key]
```

2. Search the file for references to `LOGIN_ERRORS` and replace each with calls to the new function, i.e.:
```
except ValueError:
                f.note = get_login_error('max_retries_exceeded')
```

P.S. Looks like there's a typo in one of the errors -- feel free to change the "account_locked" message to say "locked" instead of "blocked"

**Comment by etgibbs** [comment]
@rebecca-shoptaw nice work! was able to pass all the test cases on my end. committed the change to our branch.

**Review by rebecca-shoptaw — CHANGES_REQUESTED** [review]
Fantastic! Thank you!! One last thing courtesy of @cdrini -- if you could add a quick comment to the function explaining that the errors are nested in there to avoid web.ctx errors (or however you'd like to word it), that would be great and hopefully prevent future engineers from trying to switch it back to a regular dictionary.

**Comment by etgibbs** [comment]
@rebecca-shoptaw added those changes! normally I don't forget to add doc-strings ... thanks @cdrini

**Review by rebecca-shoptaw — APPROVED** [review]
Looks good to me!! Thanks so much!

File: openlibrary/plugins/upstream/account.py [hunk 1]
```diff
@@ -57,21 +57,44 @@
...
+        ),
+        "max_retries_exceeded": _(
+            'A problem occurred and we were unable to log you in.'
+        ),
+        "invalid_s3keys": _(
+            'Login attempted with invalid Internet Archive s3 credentials.'
+        ),
+        "wrong_ia_account": _(
+            'An Open Library account with this email is already linked to a different Internet Archive account. Please contact info@openlibrary.org.'
+        ),
```
Comment (jimchamp) [inline openlibrary/plugins/upstream/account.py:95]:
```suggestion
```
Error removed by #9067

File: openlibrary/plugins/upstream/account.py [hunk 2]
```diff
@@ -57,21 +57,44 @@
...
+        "ia_account_not_verified": _(
+            'Please verify your Internet Archive account before logging in'
+        ),
+        "missing_fields": _('Please fill out all fields and try again'),
+        "email_registered": _('This email is already registered'),
+        "username_registered": _('This username is already registered'),
+        "ia_login_only": _(
+            'Sorry, you must use your Internet Archive email and password to log in'
+        ),
```
Comment (jimchamp) [inline openlibrary/plugins/upstream/account.py:86]:
```suggestion
```
Recently removed by #9067

**Review by jimchamp — APPROVED** [review]
Thanks, @etgibbs!
Two of these errors were recently removed.  I'll make a commit to remove them on this branch, then will merge.

**Pull request merged**